pkg/backup/backup_compaction_test.go

```diff
@@ -383,20 +390,38 @@ func TestBackupCompactionLocalityAware(t *testing.T) {
 	var jobID jobspb.JobID
 	row.Scan(&jobID)
 	waitForSuccessfulJob(t, tc, jobID)
-	validateCompactedBackupForTables(t, db, []string{"foo"}, collectionURIs)
+	validateCompactedBackupForTables(t, db, []string{"foo"}, collectionURIs, start, end)
 }
 
 func validateCompactedBackupForTables(
-	t *testing.T, db *sqlutils.SQLRunner, tables []string, collectionURIs string,
+	t *testing.T, db *sqlutils.SQLRunner, tables []string, collectionURIs string, start, end string,
 ) {
 	t.Helper()
-	validateCompactedBackupForTablesWithOpts(t, db, tables, collectionURIs, "")
+	validateCompactedBackupForTablesWithOpts(t, db, tables, collectionURIs, start, end, "")
 }
 
 func validateCompactedBackupForTablesWithOpts(
-	t *testing.T, db *sqlutils.SQLRunner, tables []string, collectionURIs string, opts string,
+	t *testing.T,
+	db *sqlutils.SQLRunner,
+	tables []string,
+	collectionURIs string,
+	start, end string,
+	opts string,
 ) {
 	t.Helper()
+
+	// Ensure a backup exists that spans the start and end times.
+	var count int
+	showBackupQ := fmt.Sprintf(`SHOW BACKUP FROM LATEST IN (%s)`, collectionURIs)
+	if opts != "" {
+		showBackupQ += " WITH " + opts
+	}
+	db.QueryRow(t, fmt.Sprintf(`SELECT count(*) FROM [%s] 
+	WHERE start_time::DECIMAL = ('%s'::DECIMAL / 1e9) AND
```

Comment:
could you instead query for the start and end times and then do a require.Equal? Seems easier to debug